patches/@graphql-codegen+cli+1.21.6.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/node_modules/@graphql-codegen/cli/bin.js b/node_modules/@graphql-codegen/cli/bin.js
+index 1307a11..7fdc41d 100755
+--- a/node_modules/@graphql-codegen/cli/bin.js
++++ b/node_modules/@graphql-codegen/cli/bin.js
+@@ -514,7 +514,13 @@ async function loadDocuments(documentPointers, config) {
+     const loaders = [new codeFileLoader.CodeFileLoader(), new gitLoader.GitLoader(), new githubLoader.GithubLoader(), new graphqlFileLoader.GraphQLFileLoader()];
+     const loadedFromToolkit = await load.loadDocuments(documentPointers, {
+         ...defaultDocumentsLoadOptions,
+-        ignore: Object.keys(config.generates).map(p => path.join(process.cwd(), p)),
++        ignore: Object.keys(config.generates).map(p => {
++          let ignorePath = path.join(process.cwd(), p)
++          if (ignorePath.endsWith("**/*") === false) {
++            ignorePath += "**/*"
++          }
++          return ignorePath
++        }),
```

Comment:
I don't have an exact idea what is going on here and this patch probably breaks other stuff. I noticed that the following codegen config:

```yaml

generates:
  ./demo/gql/:
    preset: ./lib/preset.js
    plugins: [./lib/plugin.js]
```

Would result in the following error:

```
Error: unable to find loader for glob "/Users/laurinquast/projects/graphql-typescript-integration/demo/gql/"
        at addGlobsToLoaders (/Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-tools/load/index.js
:295:19)
        at collectPathsFromGlobs (/Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-tools/load/inde
x.js:331:5)
        at collectSources (/Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-tools/load/index.js:21
6:23)
        at loadTypedefs (/Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-tools/load/index.js:609:
21)
        at loadDocuments (/Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-codegen/cli/bin.js:515:
31)
        at /Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-codegen/cli/bin.js:916:55
        at Task.task (/Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-codegen/cli/bin.js:761:17)
    Error: unable to find loader for glob "/Users/laurinquast/projects/graphql-typescript-integration/demo/gql/"
        at addGlobsToLoaders (/Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-tools/load/index.js
:295:19)
        at collectPathsFromGlobs (/Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-tools/load/inde
x.js:331:5)
        at collectSources (/Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-tools/load/index.js:21
6:23)
        at loadTypedefs (/Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-tools/load/index.js:609:
21)
        at loadDocuments (/Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-codegen/cli/bin.js:515:
31)
        at /Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-codegen/cli/bin.js:916:55
        at Task.task (/Users/laurinquast/projects/graphql-typescript-integration/node_modules/@graphql-codegen/cli/bin.js:761:17)
```

I dug into it and it seems that the folder is not properly ignored because the string must be a glob pattern. So I simply added it there.